app/views/questionnaire.py

```diff
@@ -586,15 +586,22 @@ def _remove_dependent_answers_from_completed_blocks(answer_id, questionnaire_sto
     :param questionnaire_store: holds the completed blocks
     :return: None
     """
+    answer_in_repeating_group = g.schema.answer_is_in_repeating_group(answer_id)
     dependencies = g.schema.dependencies[answer_id]
+
     for dependency in dependencies:
+        dependency_in_repeating_group = g.schema.answer_is_in_repeating_group(dependency)
+
         answer = g.schema.get_answer(dependency)
         question = g.schema.get_question(answer['parent_id'])
         block = g.schema.get_block(question['parent_id'])
 
-        location = Location(block['parent_id'], 0, block['id'])
-        if location in questionnaire_store.completed_blocks:
-            questionnaire_store.completed_blocks.remove(location)
+        if dependency_in_repeating_group and not answer_in_repeating_group:
+            questionnaire_store.remove_completed_by_group_and_block(block['parent_id'], block['id'])
+        else:
+            location = Location(block['parent_id'], group_instance, block['id'])
+            if location in questionnaire_store.completed_blocks:
+                questionnaire_store.completed_blocks.remove(location)
```

Comment:
Since we now have an interface for removing completed blocks from the questionnaire store could we please add a second method to remove complete blocks by group id, block id + group instance? Or maybe just have a single `remove_completed_blocks` method that can accept any of these arguments?

This would be a useful refactor as we'd like to gradually try and remove this sort of logic from the views package.